src/chain.h

```diff
@@ -180,14 +180,24 @@ class CBlockIndex
     uint32_t nBits{0};
     uint32_t nNonce{0};
 
+    //! (currently memory only, but don't have to be)
+    bool cache_init;
+    uint256 cache_block_hash, cache_PoW_hash;
+
     //! (memory only) Sequential id assigned to distinguish order in which blocks are received.
     int32_t nSequenceId{0};
 
     //! (memory only) Maximum nTime in the chain up to and including this block.
     unsigned int nTimeMax{0};
 
+    void SetNull()
+    {
+        cache_init = false;
+    }
+
     CBlockIndex()
     {
+        SetNull();
```

Comment:
should check if `SetNull` went well.
```suggestion
        SetNull();
        #include <assert.h>
        assert(cache_init = false);
```